cli/analyzecli/display.go

```diff
@@ -7,143 +7,119 @@ import (
 	"os"
 	"sync"
 	"sync/atomic"
-	"time"
 
 	"github.com/brimdata/brimcap/analyzer"
-	"github.com/brimdata/zed/pkg/display"
 	"github.com/brimdata/zed/pkg/nano"
 	"github.com/brimdata/zed/pkg/units"
+	"github.com/gosuri/uilive"
 )
 
-type Display struct {
-	analyzer analyzer.Interface
-	display  interface {
-		Run()
-		Close()
-	}
-	pcapsize      int64
-	start         nano.Ts
-	span          nano.Span
-	json          bool
-	warningsCount int32
-
-	warningsMu sync.Mutex
-	warnings   map[string]int
+type Display interface {
+	Warn(string) error
+	Stats(analyzer.Stats) error
+	End()
 }
 
-func NewDisplay(json bool) *Display {
-	return &Display{
-		json:     json,
-		start:    nano.Now(),
+func NewDisplay(jsonOut bool, pcapsize int64, span nano.Span) Display {
+	if jsonOut {
+		var s *nano.Span
+		if span.Dur > 0 {
+			s = &span
+		}
+		return &jsonDisplay{
+			encoder:  json.NewEncoder(os.Stderr),
+			pcapsize: pcapsize,
+			span:     s,
+		}
+	}
+	return &statusLineDisplay{
+		live:     uilive.New(),
+		pcapsize: pcapsize,
 		warnings: make(map[string]int),
 	}
 }
 
-func (a *Display) Run(analyzer analyzer.Interface, pcapsize int64, span nano.Span) {
-	analyzer.WarningHandler(a)
-	a.analyzer = analyzer
-	a.pcapsize = pcapsize
-	a.span = span
-	if a.json {
-		interval := time.Second
-		a.display = newJSONDisplayer(a, interval)
-	} else {
-		interval := time.Millisecond * 100
-		a.display = display.New(a, interval)
-	}
-	go a.display.Run()
+type statusLineDisplay struct {
+	live          *uilive.Writer
+	pcapsize      int64
+	warnings      map[string]int
+	warningsCount int64
+	warningsMu    sync.Mutex
```

Comment:
Nit: Group to indicate that `warningsMu` only protects `warnings`.
```suggestion
	warningsCount int64

	warningsMu sync.Mutex
	warnings   map[string]int
```